lib/rubocop/cop/flexport/global_model_access_from_engine.rb

```diff
@@ -94,10 +113,22 @@ def global_model_names
           @global_model_names ||= calculate_global_models
         end
 
+        def global_factories
+          @global_factories ||= spec_factory_paths.flat_map do |path|
+            source_code = File.read(path)
+            source = RuboCop::ProcessedSource.new(source_code, RUBY_VERSION.to_f)
+            find_factories(source.ast)
+          end
+        end
```

Comment:
do we have a sense of the performance impact of loading/parsing all these files?

Reply:
nice approach! per offline discussions, i thought instances were reused, but it look lik classes are indeed reinstantiated for each inspected file:
https://github.com/maxh/rubocop/blob/master/lib/rubocop/runner.rb#L294
https://github.com/maxh/rubocop/blob/master/lib/rubocop/cop/team.rb#L42

feels a little hacky to attach this to the class, seems like rubocop should make expose ways for cops to use this facility in a more general way:

https://github.com/maxh/rubocop/blob/master/lib/rubocop/cached_data.rb#L25

but this works. probably we should do something similar for the engine api files too